src/desktop_notifier/util/pkg_metadata_helper.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+"""
+Helper methods for parsing data out of the pyproject.toml configuration in a
+hopefully future-proof way.
+"""
+from email import parser, policy
+from importlib.metadata import metadata
+
+__DESKTOP_NOTIFIER_PACKAGE_NAME__ = "desktop-notifier"
+__desktop_notifier_metadata = metadata(__DESKTOP_NOTIFIER_PACKAGE_NAME__)
+
+
+def _get_project_url() -> str | None:
+    """Parse project URLs from pyproject.toml package metadata."""
+    project_urls = __desktop_notifier_metadata.get("Project-URL")
+
+    # 'project_urls' should be a string that looks like this
+    #    "Homepage, https://github.com/samschott/desktop-notifier"
+    if isinstance(project_urls, str):
+        return project_urls.split(",", maxsplit=1)[1].strip()
+    else:
+        print(f"WARNING: Could not parse project URL from {project_urls}")
+
+
+# Using the approach in https://stackoverflow.com/questions/75801738/importlib-metadata-doesnt-appear-to-handle-the-authors-field-from-a-pyproject-t
```

Comment:
as i mentioned before i hate this... _a lot_. but sadly i have enough experience suffering through `pkg_resources`, `importlib.metadata`, `importlib_metadata`, and on and on to believe this guy when he says it's the most portable / dependency free way to go about this.